### DISCUSSION TITLE
Roadmap 2024-2025

### DISCUSSION THREAD

**Discussion by syhol** [jdx/mise#3340] [Ideas]
I'm a big fan of mise and I love what jdx had done over the last year. There's only 1 month left in 2024 and the roadmap looks nearly complete:

### Tasks
> this is the newest headline feature of mise and needs to be refined, tested, and iterated on before it can come out of experimental

:white_check_mark: No longer new. No longer experimental. It's stable. We love it.

### Documentation website
> we've outgrown what is mostly a single README

:white_check_mark: https://mise.jdx.dev/ is looking very nice today.

### Supply chain hardening
> securing mise is very important and this topic has had a lot of interest from the community. We plan to make several improvements on this front

:construction: Aqua and ubi together account for over half of the registry, asdf is nearly a minority, and cosign/slsa can be used to verify aqua tool signatures. Still a little way to go, but progress is solid.

### Improved python development
> better virtualenv integration, precompiled python binaries, and other areas are topics that frequently come up to improve

:construction: Ignoring some minor :eyes:  problems with pipx backend, the virtualenv experience is really nice, uv support is great, and precompiled python binaries are used by default. I know there is still chatter about some virtualenv handling but I think the issues are minor.

### Improved plugin development
>it's unclear what we'll do exactly but in general we want to make the experience of vending tools for asdf/mise to be better and safer.

:construction: Depends how you interpret this goal, but mise now has 6 stable backends making it easier than ever to provide tools to mise. Plugins themselves haven't changes much, but most tools no longer need plugins at all.

### GUI/TUI
> While we're all big CLI fans, it still would be great to better visualize what tools are available, what your configuration is, and other things via some kind of UI.

❔ I don't think there was much movement in this space

## 2025 Ideas...

I can't wait to see what comes next. I'm going to post my wishlist below 😁

**Comment by syhol** [comment]
My wishlist is getting increasingly smaller with every release. But it looks something like this in order of priority:

- Continued supply chain hardening
- Stability - Lots of great changes and new features recently so there are bound to be some minor snagging issues. Now I would love to see mise gain a reputation for stability and reliability.
- PHP - running multiple php versions locally today doesn't seem like a good experience, [this reddit thread tells people to just use docker](https://www.reddit.com/r/PHPhelp/comments/xw5nzx/php_version_manager/), and there was some twitter/x/youtube drama recently around how hard it is to get started with php on a mac which lead to the creation of https://php.new/ which I think provides statically compiled php binaries, so maybe thats some good infrastructure to depend on.
- pkgx backend - It's actually a pretty amazing experience for software with dynamic library dependencies, for example I can run mongod without needing to manually install any other dependencies, and it works across mac and linux. The [pkgx pantry](https://github.com/pkgxdev/pantry) seems pretty active, and it looks like it can be implemented as a backend in a similar way to how aqua was, as its just a big repo of archives. However, looking at [GitHub Star History](https://star-history.com/#pkgxdev/pkgx&jdx/mise&Date) it appears to have stalled in interest.

**Reply by jdx** [reply]
> PHP

This should probably go into vfox-php since I suspect php is more popular there. I don't see myself maintaining php in core.

> pkgx backend

There is a lot of maintenance cost to maintain backends especially with something like pkgx that I suspect is likely to change. I think mise shouldn't stretch too thin given pkgx wouldn't provide us much more than we get today with aqua.
